lua/packer.lua

```diff
@@ -796,6 +818,89 @@ packer.plugin_complete = function(lead, _, _)
   return completion_list
 end
 
+---Snapshots installed plugins
+---@param snapshot_name string @ can be an absolute path or just a filename
+packer.snapshot = function(snapshot_name)
+  local async = require('packer.async').sync
+  local await = require('packer.async').wait
+  local log = require_and_configure 'log'
+
+  async(function()
+    local snapshot = require 'packer.snapshot'
+    local snapshot_path = snapshot_name
```

Comment:
What if `snapshot_name` is nil? Can the user take an anonymous snapshot? We should check for validity either way.